### PR TITLE
feat(formula): calculate formula,get locale from user config

### PR DESCRIPTION
**Pull request by Dushusir**
- [x] calculate formula controller
- [x] get locale from user config
- [x] move search list to controller

### CODE REVIEW

File: packages/sheets-plugin-formula/package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "scripts": {},
     "dependencies": {
         "@univerjs/base-docs": "workspace:*",
+        "@univerjs/base-formula-engine": "workspace:^",
```
Comment (jikkai) [inline packages/sheets-plugin-formula/package.json:24]: :trollface:

File: packages/sheets-plugin-formula/src/commands/operations/editor-formula.operation.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export interface ISetEditorFormulaOperationParam {
 }
 
 export const SetEditorFormulaOperation: IOperation<ISetEditorFormulaOperationParam> = {
-    id: 'formula.operation.set-editor-formula-arrow',
+    id: 'formula.operation.set-editor-formula',
```
Comment (wzhudev) [inline packages/sheets-plugin-formula/src/commands/operations/editor-formula.operation.ts:15]: 感觉名字用 `SelectEditorFormluaOperation` 是不是更加贴切

File: packages/sheets-plugin-formula/src/controllers/calculate.controller.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+@OnLifecycle(LifecycleStages.Starting, CalculateController)
+export class CalculateController extends Disposable {
+    constructor(
+        @Inject(Injector) private readonly _injector: Injector,
+        @IUniverInstanceService private readonly _currentUniverService: IUniverInstanceService,
+        @IMenuService private readonly _menuService: IMenuService,
+        @ICommandService private readonly _commandService: ICommandService,
+        @IUIController private readonly _uiController: IDesktopUIController,
+        @Inject(TextSelectionManagerService) private readonly _textSelectionManagerService: TextSelectionManagerService,
```
Comment (wzhudev) [inline packages/sheets-plugin-formula/src/controllers/calculate.controller.ts:33]: 负责计算的 controller 在什么情况下会依赖到文字选择的服务呢？

Comment (wzhudev) [reply packages/sheets-plugin-formula/src/controllers/calculate.controller.ts:33]: 如果没有用到，删除掉好了

File: packages/ui-plugin-sheets/src/controllers/shortcuts/utils.ts [hunk 4]
```diff
@@ -15,3 +21,11 @@ export function whenEditorActivatedIsVisible(contextService: IContextService) {
         contextService.getContextValue(FOCUSING_EDITOR) && contextService.getContextValue(FOCUSING_EDITOR_BUT_HIDDEN)
     );
 }
+
+export function whenEditorFormulaActivated(contextService: IContextService) {
```
Comment (wzhudev) [inline packages/ui-plugin-sheets/src/controllers/shortcuts/utils.ts:25]: Renaming these functions and context keys to clarify them.

**Review by wzhudev — CHANGES_REQUESTED** [review]

File: packages/sheets-plugin-formula/src/commands/operations/editor-formula.operation.ts (hunk 2 above)
Comment (Dushusir) [reply packages/sheets-plugin-formula/src/commands/operations/editor-formula.operation.ts:15]: > 感觉名字用 `SelectEditorFormluaOperation` 是不是更加贴切

完全ok

File: packages/ui-plugin-sheets/src/controllers/shortcuts/utils.ts (hunk 4 above)
Comment (Dushusir) [reply packages/ui-plugin-sheets/src/controllers/shortcuts/utils.ts]: > Renaming these functions and context keys to clarify them.

use `whenEditorInputFormulaActivated`, `whenEditorDidNotInputFormulaActivated`, and context key `FOCUSING_EDITOR_INPUT_FORMULA`

**Review by wzhudev — APPROVED** [review]

**Pull request merged**